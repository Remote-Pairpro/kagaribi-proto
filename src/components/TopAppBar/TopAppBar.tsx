import AppBar from "@material-ui/core/AppBar";
import Button from "@material-ui/core/Button";
import IconButton from "@material-ui/core/IconButton";
import Toolbar from "@material-ui/core/Toolbar";
import Typography from "@material-ui/core/Typography";
import AccountCircle from "@material-ui/icons/AccountCircle";
import MenuIcon from "@material-ui/icons/Menu";
import React from "react";

interface ITopAppBarProps {
    authUser: firebase.User;
}

class TopAppBar extends React.Component<ITopAppBarProps> {
    constructor(appTopBarProps: ITopAppBarProps) {
        super(appTopBarProps);
    }

    public render() {
        const user = this.props.authUser;
        return (
            <AppBar position="static">
                <Toolbar>
                    <IconButton color="inherit">
                        <MenuIcon />
                    </IconButton>
                    <Typography variant="h6" color="inherit">
                        Kagaribi
                    </Typography>
                    {user ? (
                        <IconButton color="inherit">
                            <AccountCircle />
                        </IconButton>
                    ) : (
                        <Button color="inherit">Login</Button>
                    )}
                </Toolbar>
            </AppBar>
        );
    }
}

export default TopAppBar;
