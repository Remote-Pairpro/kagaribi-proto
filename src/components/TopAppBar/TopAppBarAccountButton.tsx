import IconButton from "@material-ui/core/IconButton";
import Menu from "@material-ui/core/Menu";
import MenuItem from "@material-ui/core/MenuItem";
import AccountCircle from "@material-ui/icons/AccountCircle";
import React from "react";
import { Link } from "react-router-dom";
import * as routes from "../../constants/routes";
import * as auth from "../../firebase/auth";

interface ITopAppBarAccountButtonState {
    anchorEl: any;
}

class TopAppBarAccountButton extends React.Component<
    {},
    ITopAppBarAccountButtonState
> {
    constructor(topAppBarAccountButtonProps: any) {
        super(topAppBarAccountButtonProps);
        this.state = {
            anchorEl: null,
        };
    }

    public openMenu = (event: any) => {
        this.setState({ anchorEl: event.currentTarget });
    };

    public closeMenu = () => {
        this.setState({ anchorEl: null });
    };

    public closeAndSignOut = () => {
        this.closeMenu();
        auth.signOut();
    };

    public render() {
        const anchorEl = this.state.anchorEl;
        const open = Boolean(anchorEl);

        return (
            <div>
                <IconButton
                    color="inherit"
                    onClick={this.openMenu}
                    aria-owns={open ? "menu-appbar" : undefined}
                    aria-haspopup="true"
                >
                    <AccountCircle />
                </IconButton>
                <Menu
                    id="menu-appbar"
                    anchorEl={anchorEl}
                    anchorOrigin={{
                        vertical: "top",
                        horizontal: "right",
                    }}
                    transformOrigin={{
                        vertical: "top",
                        horizontal: "right",
                    }}
                    open={open}
                    onClose={this.closeMenu}
                >
                    {/* See https://stackoverflow.com/questions/47873566/how-navigate-using-the-menuitem-material-ui-v1#comment88227691_47875510 */}
                    <MenuItem
                        component={Link}
                        {...{ to: routes.MY_PAGE }}
                        onClick={this.closeMenu}
                    >
                        My Page
                    </MenuItem>
                    <MenuItem onClick={this.closeAndSignOut}>Sign Out</MenuItem>
                </Menu>
            </div>
        );
    }
}

export default TopAppBarAccountButton;
