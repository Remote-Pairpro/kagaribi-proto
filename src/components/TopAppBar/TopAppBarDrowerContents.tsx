import List from "@material-ui/core/List";
import ListItem from "@material-ui/core/ListItem";
import ListItemIcon from "@material-ui/core/ListItemIcon";
import ListItemText from "@material-ui/core/ListItemText";
import HomeIcon from "@material-ui/icons/Home";
import SubjectIcon from "@material-ui/icons/Subject";
import React from "react";

const TopAppBarDrowerContents = () => {
    return (
        <div>
            <List>
                <ListItem button>
                    <ListItemIcon>
                        <HomeIcon />
                    </ListItemIcon>
                    <ListItemText>Home</ListItemText>
                </ListItem>
                <ListItem button>
                    <ListItemIcon>
                        <SubjectIcon />
                    </ListItemIcon>
                    <ListItemText>About</ListItemText>
                </ListItem>
            </List>
        </div>
    );
};

export default TopAppBarDrowerContents;
