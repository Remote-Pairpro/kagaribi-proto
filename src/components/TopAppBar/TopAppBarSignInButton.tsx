import Button from "@material-ui/core/Button";
import React from "react";
import { Link } from "react-router-dom";
import * as routes from "../../constants/routes";

// See https://material-ui.com/demos/buttons/#third-party-routing-library
const SignInLink = (prop: any) => <Link to={routes.SIGN_IN} {...prop} />;

class TopAppBarSignInButton extends React.Component<{}> {
    constructor(topAppBarSignInButtonProps: any) {
        super(topAppBarSignInButtonProps);
    }

    public render() {
        return (
            <div>
                <Button color="inherit" component={SignInLink}>
                    SIGN IN
                </Button>
            </div>
        );
    }
}

export default TopAppBarSignInButton;
