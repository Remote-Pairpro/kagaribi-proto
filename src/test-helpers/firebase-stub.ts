const firebaseUserDummy: firebase.User = {
    delete: null,
    emailVerified: null,
    getIdTokenResult: null,
    getIdToken: null,
    isAnonymous: null,
    linkAndRetrieveDataWithCredential: null,
    linkWithCredential: null,
    linkWithPhoneNumber: null,
    linkWithPopup: null,
    linkWithRedirect: null,
    metadata: null,
    phoneNumber: null,
    providerData: null,
    reauthenticateAndRetrieveDataWithCredential: null,
    reauthenticateWithCredential: null,
    reauthenticateWithPhoneNumber: null,
    reauthenticateWithPopup: null,
    reauthenticateWithRedirect: null,
    refreshToken: null,
    reload: null,
    sendEmailVerification: null,
    toJSON: null,
    unlink: null,
    updateEmail: null,
    updatePassword: null,
    updatePhoneNumber: null,
    updateProfile: null,
    displayName: null,
    email: null,
    photoURL: null,
    providerId: null,
    uid: null,
};

export { firebaseUserDummy };
