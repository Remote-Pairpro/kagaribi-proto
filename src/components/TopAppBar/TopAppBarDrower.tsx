import Drower from "@material-ui/core/Drawer";
import React from "react";
import TopAppBarDrowerContents from "./TopAppBarDrowerContents";

interface ITopAppBarDrowerProps {
    isDrowerOpen: boolean;
    toggleDrower: any;
}

class TopAppBarDrower extends React.Component<ITopAppBarDrowerProps> {
    constructor(topAppBarDrowerProps: ITopAppBarDrowerProps) {
        super(topAppBarDrowerProps);
    }

    public render() {
        return (
            <div>
                <Drower
                    open={this.props.isDrowerOpen}
                    onClose={this.props.toggleDrower(false)}
                >
                    <div
                        tabIndex={0}
                        role="button"
                        onClick={this.props.toggleDrower(false)}
                        onKeyDown={this.props.toggleDrower(false)}
                    >
                        <TopAppBarDrowerContents />
                    </div>
                </Drower>
            </div>
        );
    }
}

export default TopAppBarDrower;
