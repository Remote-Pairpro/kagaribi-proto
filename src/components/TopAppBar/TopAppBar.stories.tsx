import { storiesOf } from "@storybook/react";
import React from "react";
import { MemoryRouter } from "react-router";
import * as routes from "../../constants/routes";
import { firebaseUserDummy } from "../../test-helpers/firebase-stub";
import TopAppBar from "./TopAppBar";

storiesOf("TopAppBar", module)
    .addDecorator(story => (
        <MemoryRouter initialEntries={[routes.HOME, routes.SIGN_IN]}>
            {story()}
        </MemoryRouter>
    ))
    .add("Non Auth", () => {
        return <TopAppBar authUser={null} />;
    })
    .addDecorator(story => (
        <MemoryRouter initialEntries={[routes.HOME, routes.MY_PAGE]}>
            {story()}
        </MemoryRouter>
    ))
    .add("Auth", () => {
        return <TopAppBar authUser={firebaseUserDummy} />;
    });
