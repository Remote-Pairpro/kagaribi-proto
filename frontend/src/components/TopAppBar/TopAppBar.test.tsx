import { shallow } from "enzyme";
import React from "react";
import { firebaseUserDummy } from "../../test-helpers/firebase-stub";
import TopAppBar from "./TopAppBar";
import TopAppBarAccountButton from "./TopAppBarAccountButton";
import TopAppBarDrower from "./TopAppBarDrower";
import TopAppBarDrowerContents from "./TopAppBarDrowerContents";
import TopAppBarSignInButton from "./TopAppBarSignInButton";

describe("<TopAppBar />", () => {
    let user: firebase.User;
    beforeEach(() => {
        user = firebaseUserDummy;
    });

    it("should be rendered correctly", () => {
        const tree = shallow(<TopAppBar authUser={user} />);
        expect(tree).toMatchSnapshot();
    });
});

describe("<TopAppBarAccountButton />", () => {
    it("should be rendered correctly", () => {
        const tree = shallow(<TopAppBarAccountButton />);
        expect(tree).toMatchSnapshot();
    });
});

describe("<TopAppBarDrower />", () => {
    it("should be rendered correctly (opened)", () => {
        const tree = shallow(
            <TopAppBarDrower isDrowerOpen={true} toggleDrower={() => "dummy"} />
        );
        expect(tree).toMatchSnapshot();
    });

    it("should be rendered correctly (closed)", () => {
        const tree = shallow(
            <TopAppBarDrower
                isDrowerOpen={false}
                toggleDrower={() => "dummy"}
            />
        );
        expect(tree).toMatchSnapshot();
    });
});

describe("<TopAppBarDrowerContents />", () => {
    it("should be rendered correctly", () => {
        const tree = shallow(<TopAppBarDrowerContents />);
        expect(tree).toMatchSnapshot();
    });
});

describe("<TopAppBarSignInButton />", () => {
    it("should be rendered correctly", () => {
        const tree = shallow(<TopAppBarSignInButton />);
        expect(tree).toMatchSnapshot();
    });
});
