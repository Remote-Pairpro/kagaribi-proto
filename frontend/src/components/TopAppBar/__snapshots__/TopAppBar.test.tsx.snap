// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TopAppBar /> should be rendered correctly 1`] = `
<WithStyles(AppBar)
  position="static"
>
  <WithStyles(Toolbar)>
    <WithStyles(IconButton)
      color="inherit"
      onClick={[Function]}
    >
      <pure(MenuIcon) />
    </WithStyles(IconButton)>
    <TopAppBarDrower
      isDrowerOpen={false}
      toggleDrower={[Function]}
    />
    <WithStyles(Typography)
      color="inherit"
      component={[Function]}
      style={
        Object {
          "textDecoration": "none",
        }
      }
      variant="h6"
    >
      Kagaribi-proto
    </WithStyles(Typography)>
    <div
      style={
        Object {
          "marginLeft": "auto",
          "marginRight": "0",
        }
      }
    >
      <TopAppBarAccountButton />
    </div>
  </WithStyles(Toolbar)>
</WithStyles(AppBar)>
`;

exports[`<TopAppBarAccountButton /> should be rendered correctly 1`] = `
<div>
  <WithStyles(IconButton)
    aria-haspopup="true"
    color="inherit"
    onClick={[Function]}
  >
    <pure(AccountCircleIcon) />
  </WithStyles(IconButton)>
  <WithStyles(Menu)
    anchorEl={null}
    anchorOrigin={
      Object {
        "horizontal": "right",
        "vertical": "top",
      }
    }
    id="menu-appbar"
    onClose={[Function]}
    open={false}
    transformOrigin={
      Object {
        "horizontal": "right",
        "vertical": "top",
      }
    }
  >
    <WithStyles(MenuItem)
      component={[Function]}
      onClick={[Function]}
    >
      My Page
    </WithStyles(MenuItem)>
    <WithStyles(MenuItem)
      onClick={[Function]}
    >
      Sign Out
    </WithStyles(MenuItem)>
  </WithStyles(Menu)>
</div>
`;

exports[`<TopAppBarDrower /> should be rendered correctly (closed) 1`] = `
<div>
  <WithStyles(Drawer)
    onClose="dummy"
    open={false}
  >
    <div
      onClick="dummy"
      onKeyDown="dummy"
      role="button"
      tabIndex={0}
    >
      <TopAppBarDrowerContents />
    </div>
  </WithStyles(Drawer)>
</div>
`;

exports[`<TopAppBarDrower /> should be rendered correctly (opened) 1`] = `
<div>
  <WithStyles(Drawer)
    onClose="dummy"
    open={true}
  >
    <div
      onClick="dummy"
      onKeyDown="dummy"
      role="button"
      tabIndex={0}
    >
      <TopAppBarDrowerContents />
    </div>
  </WithStyles(Drawer)>
</div>
`;

exports[`<TopAppBarDrowerContents /> should be rendered correctly 1`] = `
<div>
  <WithStyles(List)>
    <WithStyles(ListItem)
      button={true}
      component={[Function]}
    >
      <WithStyles(ListItemIcon)>
        <pure(HomeIcon) />
      </WithStyles(ListItemIcon)>
      <WithStyles(ListItemText)>
        Home
      </WithStyles(ListItemText)>
    </WithStyles(ListItem)>
    <WithStyles(ListItem)
      button={true}
      component={[Function]}
    >
      <WithStyles(ListItemIcon)>
        <pure(SubjectIcon) />
      </WithStyles(ListItemIcon)>
      <WithStyles(ListItemText)>
        About
      </WithStyles(ListItemText)>
    </WithStyles(ListItem)>
  </WithStyles(List)>
</div>
`;

exports[`<TopAppBarSignInButton /> should be rendered correctly 1`] = `
<div>
  <WithStyles(Button)
    color="inherit"
    component={[Function]}
  >
    SIGN IN
  </WithStyles(Button)>
</div>
`;
